### PR TITLE
fix issue 19605 - invalid forward reference error when getting the `sizeof` of an opaque enum

### PR DESCRIPTION
**Pull request by ghost**
The default type of an opaque enum was never created (int) and additionally its specified type was not assigned to the symbol during parsing.The new field `opaque` should also help to fix 19728.

### CODE REVIEW

**Comment by Basile-z** [comment]
The header generator needs a small update too.

File: src/dmd/denum.d [hunk 1]
```diff
@@ -61,6 +61,7 @@ extern (C++) final class EnumDeclaration : ScopeDsymbol
     Expression defaultval;  // default initializer
     bool isdeprecated;
     bool added;
+    bool opaque;            // enum X;
```
Comment (jacob-carlborg) [inline src/dmd/denum.d]: All new symbols should have Ddoc comments.

**Comment by jacob-carlborg** [comment]
Should you be able to get the size of an opaque enum?

**Comment by Basile-z** [comment]
This change is not ready yet. I discover new ICE with it.

**Pull request closed without merging**